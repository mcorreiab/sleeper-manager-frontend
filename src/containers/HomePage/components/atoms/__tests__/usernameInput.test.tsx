import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import UsernameInput from "../usernameInput";

it("should render the input with the placeholder", () => {
  const tree = renderer
    .create(<UsernameInput onChange={() => {}} username="" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render the input with an username inserted", () => {
  const tree = renderer
    .create(<UsernameInput onChange={() => {}} username="username" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render the input with username missing", () => {
  const tree = renderer
    .create(
      <UsernameInput onChange={() => {}} username="" isUsernameMissing={true} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render the input with username invalid", () => {
  const tree = renderer
    .create(
      <UsernameInput onChange={() => {}} username="" isUsernameInvalid={true} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("should successfuly call on change callback when typing", () => {
  const callback = jest.fn();
  const { getByPlaceholderText } = render(
    <UsernameInput onChange={callback} username="" />
  );

  userEvent.type(getByPlaceholderText("Insert your username here"), "username");
  expect(callback).toHaveBeenCalled();
});
