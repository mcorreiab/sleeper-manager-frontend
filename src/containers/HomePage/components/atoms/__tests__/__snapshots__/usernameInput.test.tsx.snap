// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the input with an username inserted 1`] = `
Array [
  <input
    className="block w-full py-3 px-4 text-base rounded-lg border border-solid border-[#2E2E2E] outline-none focus:border-2 focus:border-solid focus:border-[#0073cd]"
    onChange={[Function]}
    placeholder="Insert your username here"
    value="username"
  />,
  <p
    className="text-[#e30202] pl-4"
  />,
]
`;

exports[`should render the input with the placeholder 1`] = `
Array [
  <input
    className="block w-full py-3 px-4 text-base rounded-lg border border-solid border-[#2E2E2E] outline-none focus:border-2 focus:border-solid focus:border-[#0073cd]"
    onChange={[Function]}
    placeholder="Insert your username here"
    value=""
  />,
  <p
    className="text-[#e30202] pl-4"
  />,
]
`;

exports[`should render the input with username invalid 1`] = `
Array [
  <input
    className="block w-full py-3 px-4 text-base rounded-lg border border-solid border-[#2E2E2E] outline-none focus:border-2 focus:border-solid focus:border-[#0073cd]"
    onChange={[Function]}
    placeholder="Insert your username here"
    value=""
  />,
  <p
    className="text-[#e30202] pl-4"
  >
    User not found
  </p>,
]
`;

exports[`should render the input with username missing 1`] = `
Array [
  <input
    className="block w-full py-3 px-4 text-base rounded-lg border border-solid border-[#2E2E2E] outline-none focus:border-2 focus:border-solid focus:border-[#0073cd] border-[#e30202]"
    onChange={[Function]}
    placeholder="Insert your username here"
    value=""
  />,
  <p
    className="text-[#e30202] pl-4"
  >
    Should inform an username
  </p>,
]
`;
